hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/db/CodecBuffer.java

```diff
@@ -153,8 +188,11 @@ private void assertRefCnt(int expected) {
     Preconditions.assertSame(expected, buf.refCnt(), "refCnt");
   }
 
-  @Override
-  protected void finalize() throws Throwable {
+  /**
+   * Provide an implementation of the {@link #finalize()} method.
+   * For performance reason, this class does not override {@link #finalize()}.
+   */
+  void finalizeImpl() {
```

Comment:
/nit can be named `detectLeaks`, no need to expose the details of finalize here.